app/src/main/java/com/devoxx/android/fragment/schedule/ScheduleMainFragment.java

```diff
@@ -176,4 +181,21 @@ public void onPageSelected(int position) {
 	public void onPageScrollStateChanged(int state) {
 		// Nothing.
 	}
+
+	// This event is used to refresh the view because the favorite status has been changed from the wearable device
+	public void onEvent(ScheduleEvent scheduleEvent) {
+		if (getActivity() == null) {
+			return;
+		}
+
+		getActivity().runOnUiThread(new Runnable() {
```

Comment:
To avoid such things "runOnUiThread" you can use android annotations feature, @UiThread annotation. I think you can annotate onEvent method.